src/emailpro/external-contact/add/emailpro-external-contact-add.controller.js

```diff
@@ -0,0 +1,110 @@
+                .getSelected()
+                .then((exchange) => {
+                    if (exchange.serverDiagnostic.version === 14 && exchange.offer === this.EmailPro.accountTypeProvider) {
+                        this.EmailProExternalContacts.getContactOptions(this.$stateParams.organization, this.$stateParams.productId)
+                            .then((data) => {
+                                this.availableMainDomains = data;
+                                this.attachOrganization2010 = this.availableMainDomains[0];
+                            })
+                            .catch((failure) => {
+                                this.$scope.resetAction();
+                                this.$scope.setMessage(this.translator.tr("exchange_tab_EXTERNAL_CONTACTS_configuration_contact_add_fail"), failure.data);
+                            });
+                    }
+                });
+
+            this.retrievingAlreadyTakenEmails();
+
+            this.$scope.submitting = () => this.submitting();
+        }
+
+        retrievingAlreadyTakenEmails () {
+            this.isLoading = true;
+
+            return this.EmailProExternalContacts
+                .retrievingContacts(this.$stateParams.productId)
+                .then((data) => {
+                    this.alreadyTakenEmails = data.list.results.map((contact) => contact.externalEmailAddress);
+                })
+                .catch((error) => {
+                    this.Alerter.alertFromSWS(this.translator.tr("exchange_tab_EXTERNAL_CONTACTS_configuration_contact_add_fail"), error, this.$scope.alerts.dashboard);
+                    this.$scope.resetAction();
+                })
+                .finally(() => {
+                    this.isLoading = false;
+                });
+        }
+
+        submitting () {
+            this.$scope.resetAction();
+
+            if (!_.isEmpty(this.attachOrganization2010)) {
+                this.model.organization2010 = this.attachOrganization2010.name;
+            }
+
+            return this.EmailProExternalContacts
+                .addingContact(this.$stateParams.productId, this.model)
+                .then(() => {
+                    this.Alerter.success(this.translator.tr("exchange_tab_EXTERNAL_CONTACTS_configuration_contact_add_success"), this.$scope.alerts.dashboard);
+                })
+                .catch((failure) => {
+                    this.Alerter.alertFromSWS(this.translator.tr("exchange_tab_EXTERNAL_CONTACTS_configuration_contact_add_fail"), failure, this.$scope.alerts.dashboard);
+                });
+        }
+
+        isCurrentEmailAlreadyTaken () {
+            return this.alreadyTakenEmails.includes(this.model.externalEmailAddress);
+        }
+
+        getEmailErrorMessage () {
+            this.externalContactAddForm.emailAddress.$setValidity("email", true);
+
+            if (this.externalContactAddForm.emailAddress.$dirty && this.externalContactAddForm.emailAddress.$error.required) {
```

Comment:
You should use ngMessages : https://docs.angularjs.org/api/ngMessages/directive/ngMessages

Reply:
Nice